js/handlers/updateHandeler.mjs

```diff
@@ -0,0 +1,41 @@
+import { makePost } from "../api/auth/API_endpoints.mjs";
+import { updatePost } from "../post/update.mjs";
+
+import { UPDATE_URL } from "../api/auth/API_endpoints.mjs";
+
+async function updatePostListener() {
+  const form = document.getElementById("edit");
+  const url = new URL(location.href);
+  console.log(url);
+  const id = url.searchParams.get("id");
+  console.log(id);
+
+  // const post = await makePost(id);
+  // console.log(post);
+  // i was trying to do as the videos says, i was trying to make async function but did not work , i have tried allot
+
+  form.addEventListener("submit", (e) => {
+    e.preventDefault();
+    console.log(e);
+
+    const form = e.target;
+    const title = form[0].value;
+    console.log(title);
+    const body = form[1].value;
+    console.log(body);
+    const image = form[2].value;
+    console.log(image);
+    const id = post.id.value;
```

Comment:
```suggestion
```

ID is defined on line 10. On this line you `overwrite` it with `undefined`

Reply:
thank i will change